apple/RNCWebView.m

```diff
@@ -402,50 +405,84 @@ - (void)didMoveToWindow
 {
   if (self.window != nil && _webView == nil) {
     WKWebViewConfiguration *wkWebViewConfig = [self setUpWkWebViewConfig];
+    NSMutableDictionary *sharedWKWebViewDictionary= [[RNCWKWebViewMapManager sharedManager] sharedWKWebViewDictionary];
+      
+    if (_keepWebViewInstanceAfterUnmount && _webViewKey != nil) {
+      WKWebView *webViewForKey = sharedWKWebViewDictionary[_webViewKey];
+      if (webViewForKey != nil) {
+        _webView = webViewForKey;
+        NSMutableDictionary *sharedRNCWebViewDictionary= [[RNCWebViewMapManager sharedManager] sharedRNCWebViewDictionary];
+        RNCWebView *rncWebView = sharedRNCWebViewDictionary[_webViewKey];
+        if (rncWebView != nil) {
+          [self removeWKWebViewFromSuperView:rncWebView];
+        }
+      }
+    }
+    
+    bool reusedWebViewInstance = _webView != nil;
+
+    if (_webView == nil) {
 #if !TARGET_OS_OSX
-    _webView = [[WKWebView alloc] initWithFrame:self.bounds configuration: wkWebViewConfig];
+      _webView = [[WKWebView alloc] initWithFrame:self.bounds configuration: wkWebViewConfig];
 #else
-    _webView = [[RNCWKWebView alloc] initWithFrame:self.bounds configuration: wkWebViewConfig];
+      _webView = [[RNCWKWebView alloc] initWithFrame:self.bounds configuration: wkWebViewConfig];
 #endif // !TARGET_OS_OSX
-
+        
+        if (_webView != nil && _webViewKey != nil) {
+          sharedWKWebViewDictionary[_webViewKey] = _webView;
+          NSMutableDictionary *sharedRNCWebViewDictionary= [[RNCWebViewMapManager sharedManager] sharedRNCWebViewDictionary];
+          sharedRNCWebViewDictionary[_webViewKey] = self;
```

Comment:
What's the respective purpose of these 2 different dictionaries?

Reply:
At a high level, the `RNCWebView`'s lifecycle is one-to-one with the lifecycle of the corresponding react-component, and the `WKWebView`'s lifecycle is not one-to-one with the lifecycle of the corresponding react-component. `sharedWKWebViewDictionary` lets us keep the `WKWebView` alive when the first react-component unmounts and when its corresponding `RNCWebView` gets destroyed.

`sharedRNCWebViewDictionary` exists because it's possible to begin with one `RNCWebView` and one `WKWebView` for a given web view key. Then it's possible to mount a second react component for the webview with the same webview key before the first react component for the first `RNCWebView` unmounts. At this point, there will be two `RNCWebView`s with the same web view key but only one `WKWebView`. In `didMoveToWindow` for the second `RNCWebView`, we need to call 

```
RNCWebView *rncWebView = sharedRNCWebViewDictionary[_webViewKey];
if (rncWebView != nil) {
  [self removeWKWebViewFromSuperView:rncWebView];
}
```

and remove the `WKWebView` from the first `RNCWebView` before adding it as a child of the second `RNCWebView`. Otherwise, the app will crash.

cc @nealmanaktola

Reply:
Oh that's interesting. Thanks for the explanation.

So, if you were you just put two WebViews on a screen with the same key, I'm guessing whichever one mounts last will "win" and the other will just be blank?

Reply:
> So, if you were you just put two WebViews on a screen with the same key, I'm guessing whichever one mounts last will "win" and the other will just be blank?

Yeah, that's what I expect to happen. Ideally, the react / JS part of the app will typically be written in a way such that there is only one web view component on screen at a time for a given key. However, I've run into this scenario with two webview components with the same key in my earlier testing, and I felt that blanking was better than crashing

Reply:
I'm in the process of updating the PR description, and I came across this other reason, which I had documented in an earlier version of the PR description, for having two maps: one for `WKWebView`s and one for `RNCWebView`s.

>  We also have a map of web view keys  to `RNCWebView`s, so we can remove the correct observer from the `WKWebView`. If we don't remove the observer when we change the `WKWebView`'s parent, the app will crash with an `EXC_BAD_ACCESS` error.

This may have been one source of crashes before we added that RNCWebView map.

cc @nealmanaktola , since we previously discussed whether we needed the two maps on iOS

Reply:
I just noticed that updated comment you added, which I appreciate.

I was wondering though, could you not check the `WKWebView`'s `superview` property for the first `RNCWebView` rather than looking it up in a dictionary?

Reply:
@TheAlmightyBob , you're right! We should be able to check the `WKWebView`'s `superview` property for the first `RNCWebView`. In these two new commits
- https://github.com/react-native-webview/react-native-webview/pull/2469/commits/d1e443333c4300eee810eb467b65d526c02dc085
- https://github.com/react-native-webview/react-native-webview/pull/2469/commits/267b40a3d2bc011d64116302e9020671c5b8acc5

, I've updated our iOS and Android implementations to (a) remove the map of RNCWebViews and (b) get the RNCWebView through the internal WebView's parent before detaching the internal WebView from its parent

Reply:
Thanks so much for linking the commits! I see the map still exists on Android, it's just used in fewer places. Was keeping it intentional, or is that change just not finished yet?